image-create-custom

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+parse_args() {
+    for arg in $@; do
+            if [[ "$arg" =~ ^rhel(-[[:digit:]]){2}(-([[:digit:]]{8}-[[:digit:]]|latest))$ ]]; then
+                image="$arg"
+                break
+            fi
+    done
+
+    if [ ! "$image" ]; then
+        echo "Missing or unsupported image (rhel-x-y-YYYYMMDD-B or rhel-x-y-latest)."
+        exit 1
+    else
+        echo "Using image '$image'"
+    fi
+}
+
+create_bootstrap() {
+    echo "#!/bin/bash
+
+BASE=\$(dirname \$0)
+\$BASE/virt-install-fedora \"\$1\" x86_64 \"$install_url\"" > $bootstrap_script
+    chmod +x $bootstrap_script
+}
+
+# parse script arguments and determine RHEL major and minor version
+parse_args "$@"
+major=${image:5:1}
+minor=${image:7:1}
+compose=${image#rhel-?-?-}
+compose=${compose//-/.}
+
+# determine image type and set compose ID accordingly
+if [[ "$image" =~ ^rhel-8-.-.{8}-.?$ ]]; then
+    compose="RHEL-$major.$minor.0-$compose"
+elif [[ "$image" =~ ^rhel-7-.-.{8}-.?$ ]]; then
+    compose="RHEL-$major.$minor-$compose"
+else
+    compose="latest-RHEL-$major.$minor"
+fi
```

Comment:
This also seems brittle -- maybe rather just let the user specify the compose ID, and name the image after that?

Reply:
I changed the script to take a `COMPOSE` (and optionally `EXTRAS_COMPOSE` for RHEL-7) environment variable, and use it to determine the repo URLs and the resulting image name.